Wire-iOS Tests/IconLabelButtonTests.swift

```diff
@@ -36,120 +36,107 @@ final class IconLabelButtonTests: XCTestCase {
-        // When
-        button.appearance = .dark(blurred: true)
-
-        // Then
-        verify(matching: button)
+    func testIconLabelButton() {
+        IconLabelButtonTestCase.Appearance.allCases.forEach {
+            verify(appearance: $0)
+        }
     }
 
-    func testIconLabelButton_Dark_Unselected_Disabled_Blurred() {
-        // When
-        button.isEnabled = false
-        button.appearance = .dark(blurred: true)
+    func verify(appearance: IconLabelButtonTestCase.Appearance, file: StaticString = #file, line: UInt = #line) {
+        button.appearance = appearance.callActionAppearance
+        button.isEnabled = appearance.isEnabled
+        button.isSelected = appearance.isSelected
 
-        // Then
-        verify(matching: button)
+        let name = "testIconLabelButton_\(appearance.description)"
+        verify(matching: button, file: file, testName: name, line: line)
     }
 
-    func testIconLabelButton_Dark_Selected_Enabled_Blurred() {
-        // When
-        button.isSelected = true
-        button.appearance = .dark(blurred: true)
-
-        // Then
-        verify(matching: button)
-    }
-
-    func testIconLabelButton_Dark_Selected_Disabled_Blurred() {
-        // When
-        button.isSelected = true
-        button.isEnabled = false
-        button.appearance = .dark(blurred: true)
-
-        // Then
-        verify(matching: button)
-    }
-
-    func testIconLabelButton_Light_Unselected_Enabled() {
-        // Given
-
-        // When
-        button.appearance = .light
+}
 
-        // Then
-        verify(matching: button)
+struct IconLabelButtonTestCase {
+
+    enum Appearance: CaseIterable {
+        typealias AllCases = [Appearance]
+
+        static var allCases: AllCases {
+            var cases = AllCases()
+
+            InteractionState.allCases.forEach { interactionState in
+                SelectionState.allCases.forEach { selectionState in
+                    BlurState.allCases.forEach { blurState in
+                        cases += [.dark(blurState, selectionState, interactionState)]
+                    }
+                    cases += [.light(selectionState, interactionState)]
+                }
+            }
+
+            return cases
+        }
+
+        case dark(BlurState, SelectionState, InteractionState)
+        case light(SelectionState, InteractionState)
+
+        var callActionAppearance: CallActionAppearance {
+            switch self {
+            case .dark(let blurState, _, _): return .dark(blurred: blurState.isBlurred)
+            case .light: return .light
+            }
+        }
+
+        var isSelected: Bool {
+            return selectionState.isSelected
+        }
+
+        var isEnabled: Bool {
+            return interactionState.isEnabled
+        }
+
+        var description: String {
+            switch self {
+            case .dark(let blurState, let selectionState, let interactionState):
```

Comment:
When you need to bind multiple associated values, you can do them all at once using a single let:

```suggestion
            case let .dark(blurState, selectionState, interactionState):
```